cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/TimePartitionedStreamFileWriter.java

```diff
@@ -88,6 +89,26 @@ protected void partitionChanged(TimePartition oldPartition, TimePartition newPar
     closePartitionWriter(oldPartition);
   }
 
+
+  /**
+   * Close the current partition if it is older than its duration.
+   *
+   * @throws IOException if fail to refresh
+   */
+  public void refresh() throws Exception {
+    if (timePartition.getStartTimestamp() == -1) {
+      // timePartition hasn'scheduleWriterRefresh been overridden yet
+      return;
+    }
+
+    // Add a 5 minute buffer, to ensure that all events for this time partition are enqueued to the writer
```

Comment:
Actually we don't need to add time. This is because this call is protected by the leader lock (notice this class is marked with not thread safe) and there won't be any unfinished write